pkg/options/options.go

```diff
@@ -89,36 +66,36 @@ func newOptions(stringOptions string, opt interface{}) interface{} {
 		t = strings.TrimPrefix(t, "*options.")
 		glog.V(3).Infof("%s options are: %s", t, string(b))
 	}()
+	setAllOptionsTo(opt, !value)
 	availableOptions := structs.Names(opt)
 	for i := range availableOptions {
 		availableOptions[i] = strings.ToLower(availableOptions[i])
 	}
 	for _, elt := range strings.Split(stringOptions, ",") {
 		switch elt {
 		case "all":
-			setAll(opt)
+			setAllOptionsTo(opt, value)
+			reflect.ValueOf(opt).Elem().FieldByName("All").SetBool(value)
+			reflect.ValueOf(opt).Elem().FieldByName("None").SetBool(!value)
```

Comment:
I'm not a huge fan of inspecting structs in this way but I do understand why you needed to do it in this case. Even if there was a way to refactor this it should probably be in a different PR, so I'll let it go for now.

Reply:
I know it's awful. I'm up for any improvement.